teos/teosd.py

```diff
@@ -68,171 +56,273 @@ def get_config(command_line_conf, data_dir):
     return config
 
 
-def main(config):
-    global db_manager, chain_monitor
+class TeosDaemon:
+    """
+    The :class:`TeosDaemon` organizes the code to initialize all the components of teos, start the service, stop and
+    teardown.
 
-    try:
-        signal(SIGINT, handle_signals)
-        signal(SIGTERM, handle_signals)
-        signal(SIGQUIT, handle_signals)
+    Args:
+        config (:obj:`dict`): the configuration object.
+
+    Attributes:
+        stop_command_event (:obj:`threading.Event`) the Event that will be set to initiate a graceful shutdown.
+        stop_event (:obj:`multiprocessing.Event`) the Event that services running on different processes will monitor
+            in order to be informed that they should shutdown.
+        block_processor (:obj:`teos.block_processor.BlockProcessor`) the BlockProcessor instance.
+        db_manager (:obj:`teos.appointments_dbm.AppointmentsDBM`) the db manager for appointments.
+        watcher (:obj:`teos.watcher.Watcher`) the `Watcher` instance.
+        chain_monitor (:obj:`teos.chain_monitor.ChainMonitor`) the ``ChainMonitor`` instance.
+        self.api_popen (:obj:`subprocess.Popen` or :obj:`None`) if set, the ``Popen`` instance running the public API.
```

Comment:
I still think these two arguments could be merged, even if they are different types.

The only point of this is to hold a process object to terminate it on teardown, it should be OK if they are slightly different to teardown since we have knowledge of what kind of server we are running (`config.get("WSGI")`).

Reply:
Done in e163fc6303fe5c4ba50d2966db1a6e94bf5896c4.